packages/open-next/src/adapters/middleware.ts

```diff
@@ -57,10 +57,19 @@ const defaultHandler = async (
           );
           return {
             type: "middleware",
-            internalEvent: result.internalEvent,
+            internalEvent: {
+              ...result.internalEvent,
+              headers: {
+                ...result.internalEvent.headers,
+                "x-opennext-initial-path": internalEvent.rawPath,
```

Comment:
I'll do a proper review later but maybe we should use `[KEY_NAME]: value` to avoid magic strings?